### PR TITLE
fix(operator): truncate sidecar pod injected label to 63 chars

### LINKED ISSUE #1031: Label "sidecar.opentelemetry.io/injected" value sometimes longer than 63 characters
With an `OpenTelemetryCollector` in `sidecar` mode, the `sidecar.opentelemetry.io/injected` label gets added to the pod using the format `[otc_namespace].[otc_name]`

https://github.com/open-telemetry/opentelemetry-operator/blob/07b9f22df8a87852464bf4877d793afcddae37c3/pkg/sidecar/pod.go#L48

Under the right conditions, the label value can exceed 63 characters, which causes Kubernetes controllers to produce the following error:

```
Error creating: Pod "[name]-59ff54b4c6-9mh4g" is invalid: metadata.labels: Invalid value: "[otc_namespace].[otc_name]": must be no more than 63 characters
```

(Replace brackets with sufficiently long values.)

If the name of the `OpenTelemetryCollector` does not identify it as such, this error message is particularly cryptic as it is not obvious what is producing the invalid label. (Took me a while to dig through various controllers and mutating webhooks to find the culprit. Though arguably this is more an issue with how Kubernetes reports such errors.)

Possible solutions:

1. Move `sidecar.opentelemetry.io/injected` to annotations
    - Annotations do not have the same restrictions on values as labels
2. Allow user to disable addition of `sidecar.opentelemetry.io/injected` label
    - Presumably this would be an option in the `OpenTelemetryCollector` resource
3. Split the `[otc_namespace]` and `[otc_name]` across multiple labels
    - This is how I've seen other projects handle such cases. Since Kubernetes object names and label values have the same restrictions, you can safely put the `[otc_namespace]` in one label and the `[otc_name]` in another, guaranteeing that they will never exceed the character limit.

Would be happy to open a PR with your preferred solution. Thanks 🙂

### PR DESCRIPTION
**Pull request by RoVernekar**
**Description:**

Truncate `sidecar.opentelemetry.io/injected` sidecar pod label to 63 characters since Kubernetes labels must be 63 characters or less.

**Link to tracking Issue:**

resolves #1031

**Testing:**

Tested with a unit test in `pod_test.go`.

**Documentation:**

### CODE REVIEW

File: pkg/sidecar/pod.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func add(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCo
 	if pod.Labels == nil {
 		pod.Labels = map[string]string{}
 	}
-	pod.Labels[label] = fmt.Sprintf("%s.%s", otelcol.Namespace, otelcol.Name)
+	pod.Labels[label] = naming.Truncate("%s.%s", 63, otelcol.Namespace, otelcol.Name)
```
[the remark this answers is not in the document]
Comment (RoVernekar) [reply pkg/sidecar/pod.go]: How about `naming.PodInstanceLabel`? The same logic is used in a few other places so I figured we could reuse this function but couldn't think of a great name.

Also I renamed `label` to `injectedLabel`. How's that?

[the remark this answers is not in the document]
Comment (jaronoff97) [reply pkg/sidecar/pod.go]: perfect :)

**Comment by jaronoff97** [comment]
@RoVernekar youll need to fill out the CLA before we can merge

**Comment by RoVernekar** [comment]
> @RoVernekar youll need to fill out the CLA before we can merge

Sorry for the delay, should be good now.

File: internal/naming/main.go [hunk 2]
```diff
@@ -124,3 +124,8 @@ func ServiceMonitor(otelcol string) string {
 func TargetAllocatorServiceAccount(otelcol string) string {
 	return DNSName(Truncate("%s-targetallocator", 63, otelcol))
 }
+
+// PodInstanceLabel returns a label value containing the namespace and instance name.
+func PodInstanceLabel(namespace string, otelcol string) string {
```
Comment (pavolloffay) [inline internal/naming/main.go:129]: is `Pod` the right prefix we want to use? Shouldn't this be just `InstanceLabel` ?

Comment (pavolloffay) [reply internal/naming/main.go:129]: I would revert other changes in this PR . The naming/main.go is used for k8s object names. 

Or expose this function in the labels.go

Comment (RoVernekar) [reply internal/naming/main.go]: Okay I went back to the original changes. Let me know if you would prefer I expose this in `labels.go` instead.

Comment (RoVernekar) [reply internal/naming/main.go]: @pavolloffay can you take another look at this when you get a chance?

**Review by yuriolisa — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**